File: TextDetector/debug/drawContours.cpp
#include <iostream>
#include <opencv2/imgcodecs.hpp>
#include <opencv2/highgui.hpp>
#include <opencv2/imgproc.hpp>

int main()
{
    cv::Mat img = cv::imread("round.png", cv::IMREAD_GRAYSCALE);
    cv::Mat thresh;
    cv::threshold(img, thresh, 127, 255, 0);
    std::vector<cv::Vec4i> hierarchy;
    std::vector<std::vector<cv::Point> > contours;
    // find contour
    cv::findContours(thresh, contours, hierarchy, CV_RETR_EXTERNAL, CV_CHAIN_APPROX_SIMPLE, cv::Point(0,0));
    // cv::drawContours(thresh, contours, 0, 127, 3);
    std::vector<cv::RotatedRect> min_rects;
    for (int i = 0; i < (int)contours.size(); i++) {
        // cv::drawContours(thresh, contours, i, 127, 3);
        min_rects.push_back(cv::minAreaRect(cv::Mat(contours[i])));
    }
    for(cv::RotatedRect rect : min_rects)
    {
        // cv::drawContours(dilated_bin_img, contours, i, 127, 1);
        cv::Point2f rect_points[4]; rect.points(rect_points);
        for(int j = 0; j < 4; j++)
        {
            std::cout << "(" << rect_points[j].y << ", " << rect_points[j].x << ")" <<  std::endl;
            cv::line(thresh, rect_points[j], rect_points[(j + 1) % 4], 127, 1, 8);
        }
    }

    // Display visualised image
    cv::namedWindow("Debug Window", cv::WINDOW_AUTOSIZE);
    cv::imshow("Debug Window", thresh);
    cv::waitKey(0);
}

File: TextDetector/debug/drawContours.py
import cv2
import numpy as np

img = cv2.imread('round.png', 0)
ret, thresh = cv2.threshold(img, 127, 255, 0)

# square_size = 8
# thresh = np.zeros((square_size, square_size), dtype=np.uint8)
# starting_point = int(square_size / 4)
# for y in range(starting_point, starting_point + int(square_size / 2)):
#     for x in range(starting_point, starting_point + int(square_size / 2)):
#         thresh[y, x] = 255

_, contours, _ = cv2.findContours(thresh, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE, offset=(0, 0))
cnt = contours[0]
# cv2.drawContours(thresh, [cnt], 0, (128,128,128), 3)
rect = cv2.minAreaRect(cnt)
# box = cv2.boxPoints(rect)
# box = np.int0(box)
# cv2.drawContours(thresh,[box],0,(127,127,127),1)
rect_points = cv2.boxPoints(rect)
print(rect_points.shape)
for j in range(4):
    cv2.line(thresh, (rect_points[j, 0], rect_points[j, 1]), (rect_points[(j + 1) % 4, 0], rect_points[(j + 1) % 4, 1]), 127, 1, 8)
print(len(contours))
# print(len(contours[0]))
# print(contours[0])

brect_x, brect_y, brect_w, brect_h = cv2.boundingRect(cnt)

# print(thresh)

cv2.imshow('thresh', thresh[brect_y:brect_y+brect_h, brect_x:brect_x+brect_w])
cv2.waitKey(0)
cv2.destroyAllWindows()